src/material/dialog/dialog.ts

```diff
@@ -384,6 +394,15 @@ export abstract class _MatDialogBase<C extends _MatDialogContainerBase> implemen
     }
   }
 
+  /**
+   * If a dialog was opened less than 300ms ago, we should consider blocking another dialog of the
+   * same type.
+   * @return true if a dialog was opened less than 300ms ago, false otherwise.
+   */
+  _shouldBlockAnotherDialogOpening(): boolean {
+    return new Date().getTime() - this._lastDialogOpenTime < 300;
```

Comment:
Where does the 300ms number come from? Currently the dialog animation is 150ms.

Also I think that this is pretty fragile. A better approach would be to have an `isAnimating` flag on the dialog container. It should be straightforward to do, because we already have `_onAnimationStart` and `_onAnimationDone` callbacks.

Reply:
- https://github.com/angular/components/pull/23085#issuecomment-871637683
- https://github.com/angular/components/pull/23085#issuecomment-872449911

Reply:
I agree and I don't prefer this naive approach, but that's what past reviews and presubmits suggested that we go with.

In https://github.com/angular/components/pull/23085/files#diff-fcade7d193ab41c7b9e2b072af496bedcf5b73ee519e237e32f8f5e00328246dR81, I had `_dialogAnimatingOpen` in `_MatDialogBase`, but you are suggesting using an attribute on the `<mat-dialog-container>` DOM element instead?

Reply:
Once focus has been moved, the keyboard events won't be a problem anymore because they'll fire from inside the dialog. We could have another flag like `_hasCapturedFocus` which is flipped inside the `focusInitialElementWhenReady` callback.

Reply:
The `_hasCapturedFocus` would be on the first dialog's `_MatDialogContainerBase`. Then when the 2nd dialog open is called and we see a matching `componentOrTemplateRef`, we would find the matching `DialogRef` in `this.openDialogs` and check it's `_containerInstance. _hasCapturedFocus`?

- If it's `false`, then we throw an error or return the `DialogRef` from the first dialog. 
- If it's `true`, then what? Are we thinking that it shouldn't be possible to get a duplicate `open` event in this case and we should go ahead with opening the second dialog (with the same `componentOrTemplateRef`)?

Reply:
Correct. If focus made it into the dialog, then it shouldn't be possible to accidentally trigger another dialog from the same trigger.

Reply:
Would this introduce a race condition that could lead to inconsistent behavior?

Reply:
Focus is moved as a result of the animation finishing so I don't think there would be a race condition.